docs/source/common/debugging.rst

```diff
@@ -133,43 +207,44 @@ See Also:
     - :func:`~pytorch_lightning.utilities.model_summary.summarize`
     - :class:`~pytorch_lightning.utilities.model_summary.ModelSummary`
 
+
 ----------------
 
-Shorten Epochs
-==============
 
-Sometimes it's helpful to only use a percentage of your training, val, test or predict data (or a set number of batches).
-For example, you can use 20% of the training set and 1% of the validation set.
+*************************************
+Debugging with Distributed Strategies
+*************************************
 
-On larger datasets like Imagenet, this can help you debug or test a few things faster than waiting for a full epoch.
+DDP Debugging
+=============
 
-.. testcode::
+If you are having a hard time debugging DDP on your remote machine you can debug DDP locally on the CPU. Note that this will not provide any speed benifits.
 
-    # use only 10% of training data and 1% of val data
-    trainer = Trainer(limit_train_batches=0.1, limit_val_batches=0.01)
+.. code-block:: python
 
-    # use 10 batches of train and 5 batches of val
-    trainer = Trainer(limit_train_batches=10, limit_val_batches=5)
+    trainer = Trainer(accelerator="cpu", strategy="ddp", devices=2)
 
-----------------
+To inspect your code, you can use `pdb <https://docs.python.org/3/library/pdb.html>`_ or `breakpoint() <https://docs.python.org/3/library/functions.html#breakpoint>`_
+or use regular print statements.
 
-Validation Sanity Check
-=======================
+.. code-block:: python
```

Comment:
lets use test code as much as possible to about syntax errors and API missalignment